File: Ahorn/entities/DarkZone.jl
module DarkZone

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/DarkZone" DarkZone(x::Integer, y::Integer, PlayerLightRadius::Number=40.0, height::Integer=64, width::Integer=64)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(DarkZone),
    "dim" => Ahorn.EntityPlacement(DarkZone)
)

function Ahorn.selection(entity::DarkZone)
    return Ahorn.getEntityRectangle(entity)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DarkZone, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.getEntityRectangle(entity), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/darkZone.lua
local darkZone = {}
darkZone.name = "MaggyHelper/DarkZone"
darkZone.depth = -50
darkZone.placements = {
    { name = "normal", data = { width = 64, height = 64, Kglobal::PlayerLightRadius = 40.0, flag = "" } },
    { name = "dim", data = { width = 64, height = 64, Kglobal::PlayerLightRadius = 60.0, flag = "" } }
}
darkZone.fieldInformation = {
    Kglobal::PlayerLightRadius = { fieldType = "number", minimumValue = 10.0 },
    flag = { fieldType = "string" }
}
darkZone.fieldOrder = { "x", "y", "width", "height", "Kglobal::PlayerLightRadius", "flag" }
return darkZone
